### PR TITLE
feat: add toggle and timed button

### PR DESCRIPTION
**Pull request by claudio-OST**
<h2>Adding New UI Components: Toggle, LikeButton, and CopyLinkButton</h2>

<h3>Overview</h3>
<p>This pull request introduces three new UI components - Toggle, LikeButton, and CopyLinkButton - designed to enhance user interactivity and visual appeal in our application. Each component is built with flexibility and customization in mind, providing a range of functionalities to fit various use cases.</p>

<h3>Features and Descriptions</h3>

<ul>
  <li>
    <strong>Toggle Component</strong>
    <ul>
      <li>A versatile and customizable toggle switch.</li>
      <li>Offers distinct styling for default, hover, and active states.</li>
      <li>Supports custom icons and active state color customization.</li>
      <li>Can be easily integrated into various parts of the application for toggle functionality.</li>
    </ul>
  </li>
  <li>
    <strong>LikeButton Component</strong>
    <ul>
      <li>An interactive button for liking content, extending the Toggle component.</li>
      <li>Incorporates a timing function to display an alternative label temporarily.</li>
      <li>Accepts an optional counter to display the number of likes.</li>
      <li>Suitable for social media interactions, comment sections, or similar platforms.</li>
    </ul>
  </li>
  <li>
    <strong>CopyLinkButton Component</strong>
    <ul>
      <li>A user-friendly button for copying links to the clipboard.</li>
      <li>Maintains a consistent icon across all states for simplicity.</li>
      <li>Provides visual feedback to enhance user experience.</li>
      <li>Ideal for content sharing, social media, and other applications where link sharing is essential.</li>
    </ul>
  </li>
</ul>

<h3>Request for Review</h3>
<p>Any feedback or suggestions for improvement would be greatly appreciated.</p>

### CODE REVIEW

File: src/components/like-button/like-button.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+interface LikeButtonProps {
+	count: number;
+	onClick: () => void;
+}
+
+const getLikesText = (count: number) => (count > 1 ? `${count} Likes` : `${count ? count + ' ' : ''}Like`);
+
+const LikeButton: React.FC<LikeButtonProps> = ({ count, onClick }) => {
+	const [isLiked, setIsLiked] = useState(false);
```
Comment (malinovic) [inline src/components/like-button/like-button.tsx:13]: Currently, when clicking on the button, nothing happens. We should probably changed that.

File: src/components/like-button/like-button.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from 'react';
+import { Toggle } from '../toggle/toggle';
+import { HeartIcon } from '../icons';
+
+interface LikeButtonProps {
+	count: number;
```
Comment (malinovic) [inline src/components/like-button/like-button.tsx:6]: There should be a possiblity on having a isLikedByUser property. the user may have liked already and then it should initially be in the liked state.

File: src/components/like-button/like-button.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	onClick: () => void;
+}
+
+const getLikesText = (count: number) => (count > 1 ? `${count} Likes` : `${count ? count + ' ' : ''}Like`);
+
+const LikeButton: React.FC<LikeButtonProps> = ({ count, onClick }) => {
```
Comment (malinovic) [inline src/components/like-button/like-button.tsx:12]: The index.ts file is missing for the like button

File: src/components/toggle/toggle.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+		isToggled: false,
+		icon: <ReplyIcon size="s" color="base" />,
+		hoveredIcon: <ReplyIcon size="s" color="primary" />,
+		toggledIcon: <ReplyIcon size="s" color="primary" filled={true} />,
+		color: 'primary',
+		onClick: () => console.log('Comment Counter Toggle clicked'),
+	},
+};
+
+export const Copy = {
```
Comment (malinovic) [inline src/components/toggle/toggle.stories.tsx:70]: The copy button is a timed button and not a toggle button (refer to the figma). Other than that, when clicking on the copy button, nothing is copied to the clipboard. It also has not the same effect when clicking on it as in the figma.

**Review by malinovic — CHANGES_REQUESTED** [review]
Hi claudio, good work. There are still some changes open. Please refer to my comments for more infos or get back to me if you have any questions. :)

File: src/components/like-button/like-button.tsx (hunk 2 above)
Comment (claudio-OST) [reply src/components/like-button/like-button.tsx:6]: @malinovic fixed

File: src/components/toggle/toggle.stories.tsx (hunk 4 above)
Comment (claudio-OST) [reply src/components/toggle/toggle.stories.tsx]: @malinovic resolved in our discussion

File: src/components/like-button/like-button.tsx (hunk 3 above)
Comment (claudio-OST) [reply src/components/like-button/like-button.tsx]: @malinovic fixed

File: src/components/icons/iconUtils.ts [hunk 5]
```diff
@@ -11,7 +11,7 @@ export const getSizeClass = (size: 's' | 'm' | 'l') => {
 	}
 };
 
-export const getColorClass = (color: 'base' | 'white' | 'primary' | 'secondary' | 'error') => {
+export const getColorClass = (color: 'base' | 'white' | 'primary' | 'secondary' | 'error' | undefined) => {
```
Comment (malinovic) [inline src/components/icons/iconUtils.ts:14]: Why do we have and undefined, color class?

Comment (claudio-OST) [reply src/components/icons/iconUtils.ts]: @malinovic fixed

**Comment by malinovic** [comment]
Hi @claudio-OST, there are some merge conflicts open. Can you please resolve this? After that, I think it is ready to be merged. Please let me know when you are through :)

**Pull request merged**